pkg/credentialprovider/plugin/config.go

```diff
@@ -98,11 +100,7 @@ func validateCredentialProviderConfig(config *kubeletconfig.CredentialProviderCo
 		if provider.APIVersion == "" {
 			allErrs = append(allErrs, field.Required(fieldPath.Child("apiVersion"), "apiVersion is required"))
 		} else if _, ok := apiVersions[provider.APIVersion]; !ok {
-			validAPIVersions := []string{}
-			for apiVersion := range apiVersions {
-				validAPIVersions = append(validAPIVersions, apiVersion)
-			}
-
+			validAPIVersions := sets.StringKeySet(apiVersions).List()
```

Comment:
is token only supported in `v1`?

Reply:
Yeah, because there isn't reason to encourage using the older version of the API, and you have to make non-trivial changes to a plugin to get it to use SA tokens as input anyway.  The warning you mentioned here https://github.com/kubernetes/kubernetes/pull/128372#discussion_r1830020878 can be ignored because we are choosing not to add the new fields in the old API versions.